pymongo/message.py

```diff
@@ -1598,8 +1602,16 @@ def __init__(self, flags, payload_document):
         self.flags = flags
         self.payload_document = payload_document
 
-    def raw_response(self, cursor_id=None):
-        raise NotImplementedError
+    def raw_response(self, cursor_id=None, user_fields={}):
+        """
+        cursor_id is ignored
+        user_fields is used to determine which fields must not be decoded
+        """
+        codec_options = DEFAULT_CODEC_OPTIONS.with_options(
+            document_class=RawBSONDocument)
+        inflated_response = _decode_selective(
+            RawBSONDocument(self.payload_document), user_fields, codec_options)
```

Comment:
Do we need this `RawBSONDocument` here?

Reply:
I don't think we need `_decode_selective` either. I think we can just do:
```
inflated_response = raw_bson._inflate_bson(self.payload_document), DEFAULT_RAW_BSON_OPTIONS)

Reply:
We do need `_decode_selective` because we need the top-level document that this returns to be a dictionary so that we can modify `nextBatch`/`firstBatch` in-place in the returned `inflated_response`. Else, we will need to perform a copy in `_convert_raw_document_lists_to_streams` to construct a `dict`.

Reply:
Yes but I think `raw_bson._inflate_bson` does what we want here in a simpler way. Does it not?